src/lib.rs

```diff
@@ -1,129 +1,191 @@
+        JsonRpcClientConnector {
             client: reqwest::Client::new(),
         }
     }
 
-    /// Create a dedicated client for querying the server via RPC API.
-    pub fn as_jsonrpc(&self) -> jsonrpc::NearJsonRpcClient {
-        jsonrpc::NearJsonRpcClient {
-            near_client: self.clone(),
-        }
-    }
-
-    /// Create a dedicated client for querying the server via HTTP API.
-    pub fn as_http(&self) -> http::NearHttpClient {
-        http::NearHttpClient {
-            near_client: self.clone(),
+    /// Method executor for the client.
+    pub async fn call<M: methods::RpcMethod>(
+        &self,
+        method: &M,
+    ) -> JsonRpcMethodCallResult<M::Result, M::Error> {
+        let (method_name, params) = (
+            M::METHOD_NAME,
+            method.params().map_err(|err| {
+                JsonRpcError::TransportError(RpcTransportError::SendError(
+                    JsonRpcTransportSendError::PayloadSerializeError(err),
+                ))
+            })?,
+        );
+        let request_payload = Message::request(method_name.to_string(), Some(params));
+        let request_payload = serde_json::to_vec(&request_payload).map_err(|err| {
+            JsonRpcError::TransportError(RpcTransportError::SendError(
+                JsonRpcTransportSendError::PayloadSerializeError(err.into()),
+            ))
+        })?;
+        let request = self
+            .client
+            .post(&self.server_addr)
+            .header("Content-Type", "application/json")
+            .body(request_payload);
+        let response = request.send().await.map_err(|err| {
+            JsonRpcError::TransportError(RpcTransportError::SendError(
+                JsonRpcTransportSendError::PayloadSendError(err),
+            ))
+        })?;
+        let response_payload = response.bytes().await.map_err(|err| {
+            JsonRpcError::TransportError(RpcTransportError::RecvError(
+                JsonRpcTransportRecvError::PayloadRecvError(err),
+            ))
+        })?;
+        let response_message = from_slice(&response_payload).map_err(|err| {
+            JsonRpcError::TransportError(RpcTransportError::RecvError(
+                JsonRpcTransportRecvError::PayloadParseError(err),
+            ))
+        })?;
+        if let Message::Response(response) = response_message {
+            let response_result = response.result.or_else(|err| {
+                let err = match if err.error_struct.is_some() {
+                    err
+                } else {
+                    loop {
+                        if let RpcError { data: Some(err), .. } = err {
+                            if let Ok(info) = serde_json::from_value::<String>(err) {
+                                break RpcError::new_internal_error(None, info);
+                            };
+                        };
+                        break RpcError::new_internal_error(None, format!("<no data>"));
+                    }
```

Comment:
Yeah, it's just the copy-paste 😅..

Alternatively, it would've been

```suggestion
                    if let RpcError { data: Some(err), .. } = err {
                        if let Ok(info) = serde_json::from_value::<String>(err) {
                            RpcError::new_internal_error(None, info)
                        } else {
                            RpcError::new_internal_error(None, format!("<no data>"))
                        }
                    } else {
                        RpcError::new_internal_error(None, format!("<no data>"))
                    }
```

or perhaps

```suggestion
                    let info;
                    match err {
                        RpcError { data: Some(err), .. } if (info = serde_json::from_value::<String>(err.clone()), info.is_ok()).1 => RpcError::new_internal_error(None, info.unwrap()),
                        _ => RpcError::new_internal_error(None, format!("<no data>"))
                    }
```

Reply:
Definitely not the second 😅, I'll copy-paste, then.